packages/shared/src/variants/keima.ts

```diff
@@ -30,6 +42,24 @@ export class Keima extends GridBaduk {
           "Second move must form a Keima (Knight's move) with the first move!",
         );
       }
+    } else if (
+      // check if there are any legal follow-up moves
+      getKeimaMoves(move, this.board).every((pos) => {
+        const boardCopy = this.board.map((x) => x);
+        // simulate a move
+        this.playMoveInternal(pos);
+        // this only checks if the follow-up move is suicidal, but it could also
+        // potentially be illegal due to ko.
```

Comment:
Ah yeah, it's a little unfortunate we can't check Ko without disturbing the state of the KoDetector.  But I think we'd need to change the interface of both KoDetector and Baduk to check Future Kos easily.